key_parsers.ml

```diff
@@ -1,6 +1,23 @@
 let try_with_asn f = try Result.Ok (f ()) with Asn.Parse_error s -> Result.Error s
 let raise_asn f = match f () with Result.Ok x -> x | Result.Error s -> Asn.parse_error s
 
+let pp_of_to_string to_string fmt x =
+  Format.pp_print_string fmt (to_string x)
+
+type oid = Asn.OID.t
+let compare_oid a b =
+  String.compare (Asn.OID.to_string a) (Asn.OID.to_string b)
+let pp_oid = pp_of_to_string Asn.OID.to_string
+
+type z = Z.t
+let compare_z = Z.compare
+let pp_z = pp_of_to_string Z.to_string
+
+type cstruct = Cstruct.t
+let compare_cstruct = Cstruct.compare
+let pp_cstruct =
+  pp_of_to_string (fun cs -> Printf.sprintf "%S" @@ Cstruct.to_string cs)
```

Comment:
Done, though `Cstruct.hexdump_to_buffer` contains some newlines, which may be what we want or not.